### ISSUE TITLE
Auto-correct `self::my_method` give `.my_method`

### ISSUE THREAD

**Issue by dinatih** [bbatsov/rubocop#4071]
## Expected behavior
with a file : 
```ruby
# my.rb
class MyClass
  def my_main_method
    self::my_method
  end
end
```
and Style/RedundantSelf, Style/ColonMethodCall cops enable
Should be corrected to `my_method` (or do nothing in this context)

## Actual behavior
**(updated)** It is corrected as `.my_method`

## Steps to reproduce the problem
Run 
```shell 
$ rubocop --auto-correct my.rb
```

## RuboCop version
```
$ rubocop -V
0.47.1 (using Parser 2.3.3.1, running on ruby 2.3.3 x86_64-darwin16)
```

**Comment by mikegee** [comment]
I'm seeing a different behavior:

```
% cat > test/test.rb
class MyClass
  def my_main_method
    self::my_method
  end
end
```
```
% bin/rubocop -a --only Style/RedundantSelf,Style/ColonMethodCall test
Inspecting 1 file
C

Offenses:

test/test.rb:3:5: C: [Corrected] Style/RedundantSelf: Redundant self detected.
    self::my_method
    ^^^^^^^^^^^^^^^
test/test.rb:3:9: C: [Corrected] Style/ColonMethodCall: Do not use :: for method calls.
    self::my_method
        ^^

1 file inspected, 2 offenses detected, 2 offenses corrected
```
```
% cat test/test.rb
class MyClass
  def my_main_method
    .my_method
  end
end
```

**Comment by dinatih** [comment]
Oh yes, I am really sorry for that, you are right.

**Comment by Drenmi** [comment]
This is a curious case, indeed. On their own, both cops do the correct thing. Doing some [puts debugging](https://tenderlovemaking.com/2016/02/05/i-am-a-puts-debuggerer.html), I get this:

```
Autocorrecting ColonMethodCall on: self::my_method
Autocorrecting RedundantSelf on: self::my_method
```

So even though autocorrection is meant to happen sequentially, `RedundantSelf` somehow has not picked up the correction made by `ColonMethodCall`.

I'm wondering if this has something to do with the usage of instance variables to hold lists of nodes in `RedundantSelf`. 🤔
